backend/src/api/mutation.rs

```diff
@@ -182,6 +183,10 @@ impl Mutation {
         // because doing it like this duplicates some work
         // like checking moderator rights, input validity, etc.
 
+        if !matches!(context.authn, AuthContext::TrustedExternal) {
```

Comment:
In this case I wouldn't use `matches!` but rather derive `PartialEq, Eq` for `AuthContext` (and likely `User`). Then you can just write `if context.authn != AuthContext::TrustedExternal`